hive-metastore/src/main/java/org/apache/iceberg/hive/HiveHadoopUtil.java

```diff
@@ -30,11 +30,12 @@ public class HiveHadoopUtil {
   private HiveHadoopUtil() {}
 
   public static String currentUser() {
+    String username = null;
     try {
-      return UserGroupInformation.getCurrentUser().getUserName();
+      username = UserGroupInformation.getCurrentUser().getShortUserName();
     } catch (IOException e) {
       LOG.warn("Failed to get Hadoop user", e);
-      return System.getProperty("user.name");
     }
+    return username != null ? username : System.getProperty("user.name");
```

Comment:
Not sure why we need to change this, i think this makes it harder to read, with another conditional operator

Reply:
+1, I also think the original code was more readable.

Reply:
> Not sure why we need to change this, i think this makes it harder to read, with another conditional operator

UserGroupInformation#getShortUserName() may return null before hadoop 2.8.2. But it rarely happens in practice.
Let me restore the original code struct.

Reply:
@zhouyifan279  Ugh, didn't realize that, yea its ugly but I guess better to be defensive in that case. 

Sorry let's just restore it to your original way then?  (if ok with everyone)

Reply:
I made an IOException("Hadoop user is null") thrown to handle the case `UserGroupInformation#getShortUserName()` returns null. 

Now the code should be both more readable and robust.